core/controllers/feedback.py

```diff
@@ -108,3 +114,78 @@ def get(self):
             'cursor': new_urlsafe_start_cursor,
             'more': more,
         })
+
+
+class SuggestionHandler(base.BaseHandler):
+    """"Handles operations relating to learner suggestions."""
+
+    PAGE_NAME_FOR_CSRF = 'editor'
+
+    @base.require_user
+    def post(self, exploration_id):
+        feedback_services.create_suggestion(
+            exploration_id,
+            self.user_id,
+            self.payload.get('exploration_version'),
+            self.payload.get('state_name'),
+            self.payload.get('suggestion_content'))
+        self.render_json(self.values)
+
+
+class SuggestionActionHandler(base.BaseHandler):
+    """"Handles actions performed on threads with suggestions."""
+
+    PAGE_NAME_FOR_CSRF = 'editor'
+    ACCEPT_ACTION = 'accept'
+    REJECT_ACTION = 'reject'
+
+    @editor.require_editor
+    def put(self, exploration_id, thread_id):
+        action = self.payload.get('action')
+        if action == self.ACCEPT_ACTION:
+            exp_services.accept_suggestion(
+                self.user_id,
+                thread_id,
+                exploration_id,
+                self.payload.get('commit_message'))
+        elif action == self.REJECT_ACTION:
+             exp_services.reject_suggestion(thread_id, exploration_id)
+        else:
+            raise self.InvalidInputException('Invalid action.')
+
+        self.render_json(self.values)
+
+
+class SuggestionListHandler(base.BaseHandler):
+    """Handles operations relating to list of threads with suggestions."""
+
+    PAGE_NAME_FOR_CSRF = 'editor'
+    OPEN_LIST_TYPE = 'open'
+    CLOSED_LIST_TYPE = 'closed'
+    ALL_LIST_TYPE = 'all'
+
+    def _string_to_bool(self, str):
+        if str == 'true':
+            return True
+        if str == 'false':
+            return False
+
+    @base.require_user
+    def get(self, exploration_id):
+        threads = None
+        list_type = self.request.get('list_type')
+        has_suggestion = self.request.get('has_suggestion')
+        if list_type == self.OPEN_LIST_TYPE:
+            threads = feedback_services.get_open_threads(
+                exploration_id, self._string_to_bool(has_suggestion))
+        elif list_type == self.CLOSED_LIST_TYPE:
+            threads = feedback_services.get_closed_threads(
+                exploration_id, self._string_to_bool(has_suggestion))
+        elif list_type == self.ALL_LIST_TYPE:
```

Comment:
It's a bit odd that this doesn't follow the style set by the previous two methods, which have an extra flag for whether the threads to be returned have suggestions or not...